### PR TITLE
Update for BigInt

### PR DESCRIPTION
**Pull request by ksh8281**

### CODE REVIEW

File: src/runtime/ValueInlines.h [hunk 1]
```diff
@@ -724,14 +724,58 @@ inline double Value::toNumber(ExecutionState& state) const
...
+        return Value(0);
+    } else if (isBoolean()) {
+        return Value(asBoolean() ? 1 : 0);
+    } else {
+        return toNumericSlowCase(state);
+    }
+}
+
+inline std::pair<Value, bool> Value::toNumericWithTypeInformation(ExecutionState& state) const // <Value, isBigInt>
+{
```
Comment (clover2123) [inline src/runtime/ValueInlines.h:756]: Operation of `toNumericWithTypeInformation` is almost same to that of `toNumeric` except that it returns with a boolean which represents if the value is BigInt or not.
I'm wondering that it is worth adding a new function for this.
Or `toNumericWithTypeInformation` could recyle the existing `toNumeric` method in itself with additional boolean return.
What do you think about it?

File: src/runtime/BigInt.cpp [hunk 2]
```diff
@@ -96,6 +109,26 @@ void BigIntData::init(VMInstance* vmInstance, const char* buf, size_t length, in
...
+}
+
+bool BigIntData::greaterThan(BigInt* b) const
+{
+    return bf_cmp(&m_data, &b->m_bf) > 0;
+}
+
+bool BigIntData::greaterThanEqual(BigInt* b) const
+{
+    return bf_cmp(&m_data, &b->m_bf) > 0;
```
[the remark this answers is not in the document]
Comment (ksh8281) [reply src/runtime/BigInt.cpp:129]: Oops my mistake

File: src/runtime/ValueInlines.h (hunk 1 above)
Comment (ksh8281) [reply src/runtime/ValueInlines.h:756]: You are right. almost source code are same but there is no way to reduce duplication.
I make toNumericWithTypeInformation for performance. almost operations after toNumberic needs isBigInt or isNumber.
I think we can remove toNumberic function

Comment (ksh8281) [reply src/runtime/ValueInlines.h:756]: Oops, both functions are used on hot spot in interpreter...

**Comment by ksh8281** [comment]
I updated pr

File: src/runtime/ValueInlines.h (hunk 1 above)
Comment (clover2123) [reply src/runtime/ValueInlines.h:726]: It looks much better :)

**Review by clover2123 — APPROVED** [review]
LGTM

**Review by bbrto21 — APPROVED** [review]
LGTM

**Pull request merged**